### DISCUSSION TITLE
the row numbers change when sorting columns

### DISCUSSION THREAD

**Discussion by imed-ghomari** [RafaelGB/obsidian-db-folder#426] [General]
### Contact Details

imeed166@gmail.com

### What happened?

The order of the rows shouldn't change when sorting columns.

https://user-images.githubusercontent.com/38974541/192090621-830623dc-220c-44e3-90f2-0ea3fc0d61f3.mov

### What platform were you using?

Desktop

### Version of the plugin

2.6.6

### Are you using the latest version of the plugin in your Obsidian vault?

- [X] I have verified that I am on the latest version

### Are you check if there is a similar issue?

- [X] I have verified that there are not similar issues

**Comment by RafaelGB** [comment]
This behaviour is intentional, the order of the ids should not affect the original query order

**Reply by imed-ghomari** [reply]
I agree, the IDs shouldn't affect the sort. But I think numbers that persist and don't change with the sort will be more useful. If IDs change with the sort, then maybe another type of number should replace the IDs. I'll leave it to you to decide if this is worth it or not.

**Reply by RafaelGB** [reply]
The ids for sorted columns is controlled internally for shadow the even rows

```ts
const backgroundColor =
    (table.getRowModel().flatRows.indexOf(row) + 1) % 2
      ? StyleVariables.BACKGROUND_PRIMARY
      : StyleVariables.BACKGROUND_SECONDARY;
```

How about show that id when you hover the row context button?

**Reply by imed-ghomari** [reply]
So, show the number only when you hover over the row menu? sure, that's an option.
